src/qibo/tests/test_quantum_info.py

```diff
@@ -0,0 +1,184 @@
+    backend.assert_allclose(trace_distance(state, target), 0.0)
+
+    state = np.asarray([0.0, 1.0, 0.0, 0.0])
+    target = np.asarray([1.0, 0.0, 0.0, 0.0])
+    backend.assert_allclose(trace_distance(state, target), 1.0)
+
+
+def test_hilbert_schmidt_distance(backend):
+    with pytest.raises(TypeError):
+        state = np.random.rand(2, 2)
+        target = np.random.rand(4, 4)
+        hilbert_schmidt_distance(state, target)
+    with pytest.raises(TypeError):
+        state = np.random.rand(2, 2, 2)
+        target = np.random.rand(2, 2, 2)
+        hilbert_schmidt_distance(state, target)
+    with pytest.raises(TypeError):
+        state = np.asarray([])
+        target = np.asarray([])
+        hilbert_schmidt_distance(state, target)
+
+    state = np.asarray([1.0, 0.0, 0.0, 0.0])
+    target = np.asarray([1.0, 0.0, 0.0, 0.0])
+    backend.assert_allclose(hilbert_schmidt_distance(state, target), 0.0)
+
+    state = np.outer(np.conj(state), state)
+    target = np.outer(np.conj(target), target)
+    backend.assert_allclose(hilbert_schmidt_distance(state, target), 0.0)
+
+    state = np.asarray([0.0, 1.0, 0.0, 0.0])
+    target = np.asarray([1.0, 0.0, 0.0, 0.0])
+    backend.assert_allclose(hilbert_schmidt_distance(state, target), 2.0)
+
+
+def test_fidelity(backend):
+    with pytest.raises(TypeError):
+        state = np.random.rand(2, 2)
+        target = np.random.rand(4, 4)
+        fidelity(state, target)
+    with pytest.raises(TypeError):
+        state = np.random.rand(2, 2, 2)
+        target = np.random.rand(2, 2, 2)
+        fidelity(state, target)
+    with pytest.raises(ValueError):
+        state = np.random.rand(2, 2)
+        target = np.random.rand(2, 2)
+        fidelity(state, target, validate=True)
+
+    state = np.asarray([0.0, 0.0, 0.0, 1.0])
+    target = np.asarray([0.0, 0.0, 0.0, 1.0])
+    backend.assert_allclose(fidelity(state, target), 1.0)
+
+    state = np.outer(np.conj(state), state)
+    target = np.outer(np.conj(target), target)
+    backend.assert_allclose(fidelity(state, target), 1.0)
+
+    state = np.asarray([0.0, 1.0, 0.0, 0.0])
+    target = np.asarray([0.0, 0.0, 0.0, 1.0])
+    backend.assert_allclose(fidelity(state, target), 0.0)
+
+
+def test_process_fidelity(backend):
+    d = 2
+    with pytest.raises(TypeError):
+        channel = np.random.rand(d**2, d**2)
+        target = np.random.rand(d**2, d**2, 1)
+        process_fidelity(channel, target)
+    with pytest.raises(TypeError):
+        channel = np.random.rand(d**2, d**2)
+        process_fidelity(channel, validate=True)
+    with pytest.raises(TypeError):
+        channel = np.random.rand(d**2, d**2)
+        target = np.random.rand(d**2, d**2)
+        process_fidelity(channel, target, validate=True)
+
+    channel = np.eye(d**2)
+    backend.assert_allclose(process_fidelity(channel), 1.0)
+    backend.assert_allclose(process_fidelity(channel, channel), 1.0)
+
+
+def test_average_fidelity():
+    average_gate_fidelity(np.eye(4))
```

Comment:
I saw that the `average_gate_fidelity` differs from `process_fidelity` only by a rescaling. How about removing this test and checking both in the test above, where you can also assert both results?

Reply:
Done.